### PR TITLE
526 severance date change cv

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
This PR changes the separation / severance pay date to capture just the year.

## Testing done
Manually tested.
Unit tests written and updated.

## Screenshots
image

image

image

## Acceptance criteria
- [x] The separation or severance pay date captures only a year
- [x] The field rejects invalid responses

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      };
+      isValidYear(err, 'asdf');
+      expect(err.addError.called).to.be.true;
+    });
+
+    it('should add an error if the year contains more than just four digits', () => {
+      const err = {
+        addError: sinon.spy(),
+      };
+      isValidYear(err, '1990asdf');
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js]: I assume this also fails if you test with >4 *digits*, right?

File: src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    it('should add an error if the year is more than 3000', () => {
+      const err = {
+        addError: sinon.spy(),
+      };
+      isValidYear(err, '3001');
+      expect(err.addError.called).to.be.true;
+    });
+
+    it('should not add an error if the year is between 1900 and 3000', () => {
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js]: Seems kind of broad. Are you sure that the separation pay date doesn't have to be in the past?

File: src/applications/disability-benefits/all-claims/validations.js [hunk 3]
```diff
@@ -143,3 +143,11 @@ export const isInFuture = (err, fieldData) => {
     err.addError('Start date must be in the future');
   }
 };
+
+export const isValidYear = (err, fieldData) => {
+  const parsedInt = Number.parseInt(fieldData, 10);
+
+  if (!/^\d{4}$/.test(fieldData) || parsedInt < 1900 || parsedInt > 3000) {
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/validations.js]: Mentioned already, but not sure if severance date needs to be in the past.

**Review by markgreenburg — APPROVED** [review]

**Comment by markgreenburg** [comment]
Looks good. Just want to double check the age validation for severance pay - seems like it could need to be in the past?

File: src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js (hunk 1 above)
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js]: If it's > 4 numerical digits, it'll be > 3000 :wink:
So yes. That should be taken care of in another test.

File: src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js (hunk 2 above)
Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/tests/validations.unit.spec.js]: I...am not sure. That would make sense to me, but it wasn't in the requirements.

@djmunoz1 Can you confirm (or ask whomever can confirm)? Should the separation pay date have a year that's either the current year or in the past? Should it also be after the first period of service start date or the last PoS end date?

**Comment by cvalarida** [comment]
Yeah, good call about the date. :+1: I'll hold off on merging until we can get an answer.

**Comment by djmunoz1** [comment]
the date only has to be in the past, no further validation required on this pass.

**Comment by cvalarida** [comment]
image
@peggygannon Can I get a copy edit on the error message?

**Comment by peggygannon** [comment]
@cvalarida just one small tweak: 
The year can't be in the future

**Pull request merged**